### PR TITLE
KSTAT-441 Add initial prototyping section

### PR DESCRIPTION
**Pull request by RobLoach**
#441

### CODE REVIEW

File: docs/4-Prototyping.md [hunk 1]
```diff
@@ -1,29 +1,172 @@
...
+
+A website prototype can be any mock-up or demo of what a website could look like when it goes live. A living prototype is a prototype that's actively used throughout the lifecycle of the website. Even after the site is deployed, the prototype lives on as an active demonstration of what the website looks like.
+
+The following features and conventions come together to allow the creation of a living prototype.
+
+## Conventions
+
+### Meta-Data
+
+Meta-Data is general data that helps describe the content you're presenting. Depending on the scope, meta-data can exist almost anywhere.
```
Comment (soniktrooth) [inline docs/4-Prototyping.md]: image

File: docs/4-Prototyping.md [hunk 2]
```diff
@@ -1,29 +1,172 @@
...
+Nora Roberts:
+  birthdate: 1950
+---
+
+This list of authors is loaded into a global metadata variable named "authors".
+```
+
+### Transformers
+
+Naming a file by a [JSTransformer](https://github.com/jstransformers/jstransformer) name will have it process with the engine. The following is an example of using [Twig](http://twig.sensiolabs.org):
```
Comment (soniktrooth) [inline docs/4-Prototyping.md]: @RobLoach what's the point of transformers? Is it to enable you to use other templating engines? I think an explanation of what Kalastatic uses transformers for would be useful.

File: docs/4-Prototyping.md [hunk 3]
```diff
@@ -1,29 +1,172 @@
...
+- [twig](http://npm.im/jstransformer-twig)
+- [browserify](http://npm.im/jstransformer-browserify)
+- [scss](http://npm.im/jstransformer-scss)
+- [md](http://npm.im/jstransformer-commonmark)
+
+### Passing in environment variables
+### Meta-Data Files
+
+The [metalsmith-metadata-files](https://github.com/kalamuna/metalsmith-metadata-files) plugin will inject sister JSON files into the main file.
+
```
Comment (soniktrooth) [inline docs/4-Prototyping.md]: k but y tho?

Comment (soniktrooth) [inline docs/4-Prototyping.md]: imgres-1

File: docs/4-Prototyping.md [hunk 4]
```diff
@@ -1,29 +1,172 @@
...
+```
+<h1>Linux Torvalds</h1>
+```
+
+### Globals
+- kalastatic.yml
+
+### Assets
+
+The [metalsmith-assets-convention](https://github.com/robloach/metalsmith-assets-convention) plugin will deploy any local assets you would like to copy out through the use of `.assets` files:
```
Comment (soniktrooth) [inline docs/4-Prototyping.md]: Maybe expand a little on why you need to do this? e.g. so you can pull in a js library via npm and then use it in your site.

File: docs/4-Prototyping.md [hunk 5]
```diff
@@ -1,29 +1,170 @@
...
+*example.html*
+``` html
+---
+name: Linus Torvalds
+languages:
+- PHP
+- Perl
+- Python
+---
+<h1>Linus Torvalds</h1>
```
Comment (andrewmallis) [inline docs/4-Prototyping.md]: I was a little confused to see that the variables are not used in the output. What's the point of the front matter then?

**Comment by andrewmallis** [comment]
This is coming along really nicely. Keep up the great work!

File: docs/4-Prototyping.md (hunk 5 above)
Comment (RobLoach) [inline docs/4-Prototyping.md]: This example is just using HTML (`example.html`). HTML doesn't have variables.

We can change this example to Twig though.

File: docs/4-Prototyping.md [hunk 6]
```diff
@@ -1,33 +1,229 @@
...
+      scss:
+        includePaths: [
+          "node_modules/bootstrap-sass/assets/stylesheets",
+        ]
+```
+
+3. Use the new components in a SASS file
 
+```
+@import "bootstrap/buttons";
```
Comment (RobLoach) [inline docs/4-Prototyping.md]: @soniktrooth You could probably come up with a better example here. Thoughts?

**Pull request merged**